src/lib/sfge/BaseSfgeEngine.ts

```diff
@@ -120,97 +126,77 @@ export class SfgeEngine extends AbstractRuleEngine {
 	/**
 	 * Helps make decision to run an engine or not based on the Rules, Target paths, and the Engine Options selected per
 	 * run. At this point, filtering should have already happened.
+	 * @override
 	 */
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	public shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string,string>): boolean {
-		// If the engine isn't filtered out, there's no reason to not run it.
-		return true;
+	public shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): boolean {
+		// If the EngineOptions doesn't include an SFGE config, don't try to run this engine.
```

Comment:
The new validation in `run.ts` prevents SFGE from being requested if `--projectdir` is absent. This is for if SFGE is merely default-enabled and the flag is missing. We just skip it and log a warning instead of halting with an error.

Reply:
I like the intention for the distinction, but wondering if the inconsistency could be more confusing. 
If we were to compare this with eslint-typescript which needs ts-config parameter in most cases (unless there's a tsconfig in the current directory), wouldn't we throw the error of missing tsconfig even if eslint-typescript is enabled by default?
Since sfge would be disabled by default, and the user needs to intentionally set it to enabled, I feel it's fair to be consistent and throw the error if --projectdir is missing no matter how it was invoked.

Reply:
We could postpone the check as far as to see if we have files matching the pattern sfge looks for. I believe that's how the check works for eslint-typescript as well. If no files match, SFCA already knows that this engine will not be run, and no point in checking for missing parameters.